homeassistant/components/sharkiq/vacuum.py

```diff
@@ -198,6 +217,35 @@ async def async_locate(self, **kwargs: Any) -> None:
         """Cause the device to generate a loud chirp."""
         await self.sharkiq.async_find_device()
 
+    async def async_clean_room(self, rooms: list[str], **kwargs: Any) -> None:
+        """Clean a specific room."""
+        if len(rooms) == 0:
+            raise exceptions.HomeAssistantError("No rooms to clean were provided.")
+        rooms_to_clean = []
+        valid_rooms = []
+        if self.available_rooms is not None:
+            valid_rooms = self.available_rooms
+        all_rooms_reachable = True
+        for room in rooms:
+            if room in valid_rooms:
+                rooms_to_clean.append(room)
+            elif room.capitalize() in valid_rooms:
+                rooms_to_clean.append(room.capitalize())
```

Comment:
As already @epenet commented, I would remove this code. If you want the `rooms` to be more flexible, I suggest you use something different than `areas`.

Reply:
I got rid of this logic.  do you have a recommendation for a better way of doing it?  It would be AWESOME to give users a dropdown list of rooms available in self.available_rooms to pick from, but didn't immediately see a way to do that in the docs.   I'd love to follow up this PR with another to clean up room selection to be better.